airbyte-integrations/connectors/source-amazon-ads/source_amazon_ads/manifest.yaml

```diff
@@ -1167,6 +1183,1001 @@ definitions:
+                  "salesOtherSku7d",
+                  "salesOtherSku14d",
+                  "salesOtherSku30d",
+                ]
+              reportTypeId: spPurchasedProduct
+              filters: []
+              timeUnit: SUMMARY
+              format: GZIP_JSON
+    sponsored_products_asins_targets_report_stream:
+      type: DeclarativeStream
+      name: sponsored_products_asins_targets_report_stream
+      primary_key: ["profileId", "reportDate", "adGroupName", "advertisedAsin"]
+      incremental_sync:
+        $ref: "#/definitions/incremental_sync_report_datetime_cursor"
+      transformations:
+        $ref: "#/definitions/transformation_report_add_fields"
+      retriever:
+        $ref: "#/definitions/basic_async_retriever"
+        creation_requester:
+          $ref: "#/definitions/basic_async_retriever/creation_requester"
+          request_body_json:
+            name: "asins_targets report {{ stream_slice.start_time }}"
+            startDate: "{{ stream_slice.start_time }}"
+            endDate: "{{ stream_slice.end_time }}"
+            configuration:
+              adProduct: SPONSORED_PRODUCTS
+              groupBy: ["asin"]
+              columns:
+                [
+                  "campaignName",
+                  "campaignId",
+                  "adGroupName",
+                  "adGroupId",
+                  "advertisedAsin",
+                  "purchasedAsin",
+                  "advertisedSku",
+                  "campaignBudgetCurrencyCode",
+                  "matchType",
+                  "unitsSoldClicks1d",
+                  "unitsSoldClicks7d",
+                  "unitsSoldClicks14d",
+                  "unitsSoldClicks30d",
+                  "unitsSoldOtherSku1d",
+                  "unitsSoldOtherSku7d",
+                  "unitsSoldOtherSku14d",
+                  "unitsSoldOtherSku30d",
+                  "salesOtherSku1d",
+                  "salesOtherSku7d",
+                  "salesOtherSku14d",
+                  "salesOtherSku30d",
+                  "keywordId",
+                  "targeting",
+                  "keywordType",
+                ]
+              reportTypeId: spPurchasedProduct
+              filters: []
+              timeUnit: SUMMARY
+              format: GZIP_JSON
+
+  incremental_sync_report_datetime_cursor:
+    type: DatetimeBasedCursor
+    cursor_field: reportDate
+    cursor_datetime_formats:
+      - "%Y-%m-%d"
+    datetime_format: "%Y-%m-%d"
+    lookback_window: P{{ config.get('look_back_window', 3) }}D
+    start_datetime:
+      type: MinMaxDatetime
+      datetime: "{{ config.get('start_date', today_utc().strftime('%Y-%m-%d') ) }}"
+      datetime_format: "%Y-%m-%d"
+    end_datetime:
+      datetime: "{{ today_utc().strftime('%Y-%m-%d') }}"
+      datetime_format: "%Y-%m-%d"
+    cursor_granularity: "P1D"
+    step: "P{{ config.get('events_time_step', 30) }}D"
+    start_time_option:
+      field_name: startDate
+      inject_into: body_json
```

Comment:
Is there a reason why we set up the `DatetimeBasedCursor` to inject `startDate` and `endDate` into the body_json here? But I also see `startDate: "{{ stream_slice.start_time }}"` used in a lot of async retriever json body as well. Is it because we don't inject these slice values into the `creation_requester`?

If this is needed then let's add a comment as to why we must define it here. But if it's unneeded then can we remove it? I would also say that having the `DatetimeBasedCursor` inject options is the preferred way of doing things over our own interpolation in the retriever/requester when possible

Reply:
datetimebasedcursor injection works correctly with the creation_requester, but [HTTPmatcher checks](https://github.com/airbytehq/airbyte-python-cdk/blob/f62c91eccfa658f529b416031ba3cc5ece93c9c6/airbyte_cdk/test/mock_http/request.py#L51) bytes for jsonbody and we cannot guarantee the order inside it.

That's why I added start/end time directly to the requester.